src/System.Private.CoreLib/shared/System/Threading/Mutex.Windows.cs

```diff
@@ -137,42 +72,47 @@ private static OpenExistingResult OpenExistingWorker(string name, out Mutex resu
                 throw new ArgumentException(SR.Argument_EmptyName, nameof(name));
             }
 
-#if !PLATFORM_UNIX
             VerifyNameForCreate(name);
```

Comment:
Since you have enabled `Argument_WaitHandleNameTooLong` on the error path uncoditionally, I think you can delete the `VerifyNameForCreate` function now.